apps/src/calc/inputIterator.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Given a set of values (i.e. [1,2,3], and a number of parameters, generates
+ * all possible combinations of values.
+ */
+var InputIterator = function (values, numParams) {
```

Comment:
This actually seems sufficiently generic to be a `PermutationGenerator` or `PermutationIterator` and not need any references to the Input and Params concepts.

maybe s/`numParams`/`valuesPerChoice`